formats/onenote.ts

```diff
@@ -0,0 +1,345 @@
+			// All notebooks have sections but not all notebooks have section groups
+			let sectionGroups: SectionGroup[] | undefined | null = notebook?.sectionGroups;
+
+			this.createSectionList(sections);
+			sectionGroups?.forEach((sectionGroup) => {
+				this.modal.contentEl.createEl('h6', {
+					text: sectionGroup.displayName!,
+				});
+				this.createSectionList(sectionGroup.sections!);
+			});
+		});
+	}
+
+	createSectionList(sections: OnenoteSection[]) {
+		const contentArea: HTMLDivElement = this.modal.contentEl.querySelector('.contentArea')!;
+
+		sections?.forEach((section) => {
+			let label = contentArea.createEl('label');
+			let checkbox = label.createEl('input');
+			label.appendChild(document.createTextNode(section.displayName!));
+			checkbox.type = 'checkbox';
+			label.createEl('br');
+
+			// Add/remove a section from this.selectedSections
+			checkbox.addEventListener('change', () => {
+				if (checkbox.checked) this.selectedSections.push(section);
+				else {
+					const index = this.selectedSections.findIndex((sec) => sec.id === section.id);
+					if (index !== -1) {
+						this.selectedSections.splice(index, 1);
+					}
+				}
+			});
+		});
+	}
+
+	async import(progress: ProgressReporter): Promise<void> {
+		let outputFolder = await this.getOutputFolder();
+		if (!outputFolder) {
+			new Notice('Please select a location to export to.');
+			return;
+		}
+
+		for (let section of this.selectedSections) {
+			let notebookFolder: TFolder = await this.createFolders(outputFolder.path + '/' + section.parentNotebook?.displayName);
+			let sectionFolder: TFolder = await this.createFolders(notebookFolder.path + '/' + section.displayName);
+
+			let pages: OnenotePage[] = (await (await fetch(`
+					https://graph.microsoft.com/v1.0/me/onenote/sections/${section.id}/pages?$select=id,title,createdDateTime,lastModifiedDateTime`,
+			this.requestParams)).json()).value;
+
+			pages.forEach(async (page) =>
+				this.processFile(progress, sectionFolder, await (await fetch(
+					`https://graph.microsoft.com/v1.0/me/onenote/pages/${page.id}/content?includeInkML=true`,
+					this.requestParams)).text(), page));
+		}
+	}
+
+	async processFile(progress: ProgressReporter, folder: TFolder, content: string, page: OnenotePage) {
+		try {
+			const splitContent = this.convertFormat(content);
+
+			let parsedPage = parseHTML(splitContent.html);
+			parsedPage = this.convertInternalLinks(parsedPage);
+			parsedPage = this.convertTags(parsedPage);
+			parsedPage = this.getAllAttachments(parsedPage);
+
+			let mdContent = htmlToMarkdown(parsedPage).trim();
+			const fileRef = await this.saveAsMarkdownFile(folder, page.title!, mdContent);
+
+			await this.fetchAttachmentQueue(progress, fileRef);
+
+			// Add the last modified and creation time metadata
+			const writeOptions: DataWriteOptions = {
+				ctime: Date.parse(page.createdDateTime!.toString()) ||
+					   Date.parse(page.lastModifiedDateTime!.toString()) ||
+					   Date.now(),
```

Comment:
if createdDateTime and lastModifiedDateTime are both not null, is `Date.now()` fallback needed?